mypy/checker.py

```diff
@@ -984,7 +984,12 @@ def is_unannotated_any(t: Type) -> bool:
         check_incomplete_defs = self.options.disallow_incomplete_defs and has_explicit_annotation
         if show_untyped and (self.options.disallow_untyped_defs or check_incomplete_defs):
             if fdef.type is None and self.options.disallow_untyped_defs:
-                self.fail(message_registry.FUNCTION_TYPE_EXPECTED, fdef)
+                if (not len(fdef.arguments) or
+                        (len(fdef.arg_names) == 1 and 'self' in fdef.arg_names)):
+                    self.fail(message_registry.RETURN_TYPE_EXPECTED_DISALLOW_UNTYPED, fdef)
+                    self.note('Use "-> None" if function does not return a value', fdef)
```

Comment:
I would only show this note if the function body doesn't have non-trivial return statements (i.e. `return x` not just plain `return`).

Reply:
okay sounds good-- could you point me to how to find if there's a return value?

Reply:
You probably have to create a new `StatementVisitor` that traverses `fdef.body` looking for any `ReturnStmt` instances whose `expr` is not `None`.

(And don't take my word for it. :-)

Reply:
You can use `srubgen.has_return_statement()` (Maybe also move it to some common file like `nodes.py`?)

Reply:
(Fix: srubgen -> stubgen)

Reply:
I took a stab at implementing this in `nodes.py`... let me know if it's reasonable! 😬